vivisect/analysis/generic/symswitchcase.py

```diff
@@ -0,0 +1,1307 @@
+    * Second, to wire things up.
+    
+    Tracks the state of analysis so each component can easily do it's part 
+    without handing a ton of variables around between functions.
+    '''
+    def __init__(self, vw, jmpva, timeout=180):
+        self.vw = vw
+        self.jmpva = jmpva
+        self.timeout = timeout
+        self.jmpop = vw.parseOpcode(jmpva)
+        logger.info('=== analyzing SwitchCase branch at 0x%x: %r ===' % (jmpva, self.jmpop))
+
+        self.sctx = vs_anal.getSymbolikAnalysisContext(vw)
+        self.xlate = self.sctx.getTranslator()
+
+        # 32-bit i386 thunk_reg handling.  this should be the only oddity for this architecture
+        for tva, regname, tgtval in vw.getVaSetRows('thunk_reg'):
+            fname = vw.getName(tva, True)
+            trobj = ThunkReg(regname, tgtval)
+            self.sctx.addSymFuncCallback(fname, trobj.thunk_reg)
+            logger.debug( "sctx.addSymFuncCallback(%s, thunk_reg)" % fname)
+
+
+        self._sgraph = None
+        self._codepath = None
+        self._codepathgen = None
+
+
+        try:
+            self.max_instr_count = vw.config.viv.analysis.symswitchcase.max_instr_count
+            self.max_cases = vw.config.viv.analysis.symswitchcase.max_cases
+            self.case_failure = vw.config.viv.analysis.symswitchcase.case_failure
+            self.min_func_instr_size = vw.config.viv.analysis.symswitchcase.min_func_instr_size
+
+        except AttributeError as e:
+            logger.warning('configuration failure, using defaults: %r', e)
+            self.max_instr_count  = 10
+            self.max_cases   = 500
+            self.case_failure = 5000
+            self.min_func_instr_size = 10
+
+        self.clearCache()
+
+
+
+    def clearCache(self):
+        '''
+        While analyzing a switchcase, we end up caching several calculated items.
+        This function clears them, so we can reset the state, like to skip 
+        inappropriate paths (eg.   if x> 3 and if x == 0
+        '''
+        self.cspath = None
+        self.aspath = None
+        self.fullpath = None
+
+        self.upper = None
+        self.lower = None
+        self.count = None
+        self.baseoff = None
+        self.baseIdx = None
+
+        self.jmpsymvar = None
+
+
+    def analyze(self):
+        '''
+        Generic "analyze" function, which simply calls makeSwitch()
+        '''
+        self.makeSwitch()
+
+    
+    
+    #### primitives for switchcase analysis ####
+    def getJmpSymVar(self):
+        '''
+        returns the Simple Symbolik state of the dynamic target of the branch/jmp
+        this is the symbolik register at the start of the last codeblock.
+        '''
+        if self.jmpsymvar is not None:
+            return self.jmpsymvar
+
+        # TODO: this is not always the right operand for all architectures.  improve
+        self.jmpsymvar = self.xlate.getOperObj(self.jmpop, 0)
+        return self.jmpsymvar
+
+    def getSymTarget(self, short=True):
```

Comment:
Who uses the short param?

Reply:
it looks like i removed the programmatic need for it.  i remember using it myself, but i don't remember which code used it.